pkg/operator/sync.go

```diff
@@ -41,7 +43,16 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 	}
 
 	if config.Controllers.TerminationHandler != clusterAPIControllerNoOp {
-
+		if err := optr.syncTerminationHandler(config); err != nil {
```

Comment:
can we make this call within `syncClusterAPIController` so we make sure we wrap all the logic only once to report status appropriately based on whatever errors are reported from the inner calls?

Reply:
Also within there may be invert the logic to 1 - error earlier, 2 - logging no Op scenario i.e:`
```
if config.Controllers.Provider == config.Controllers.TerminationHandler != clusterAPIControllerNoOp {
    log mao will no op
    return nil
}

logic goes here
```

Reply:
Can do the return earlier with NoOp, that makes sense, though the comparison above doesn't tell us that the `Provider` or `TerminationHandler` aren't `NoOp`, would just tell us that one of them isn't right? We'd still have to check later?

I'd be tempted to just check the `Provider` is `NoOp`, return early, otherwise continue and then check later that `TerminationHandler` is or isn't and run appropriately

Reply:
sorry that was a typo  I meant
```if config.Controllers.TerminationHandler == clusterAPIControllerNoOp then return```

We should probably for the same for line 32.